staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -190,6 +190,21 @@ const (
 	initialBufferSize = 1024
 )
 
+// WaitForCacheSyncWrapped is a wrapper around WaitForCacheSync that generates log messages
+// indicating that the controller identified by controllerName is waiting for syncs, followed by
+// either a successful or failed sync.
+func WaitForCacheSyncWrapped(controllerName string, stopCh <-chan struct{}, cacheSyncs ...InformerSynced) bool {
+	klog.Infof("Waiting for caches to sync for %s", controllerName)
+
+	if !WaitForCacheSync(stopCh, cacheSyncs...) {
+		utilruntime.HandleError(fmt.Errorf("unable to sync caches for %s", controllerName))
+		return false
+	}
+
+	klog.Infof("Caches are synced for %s controller", controllerName)
+	return true
+}
+
 // WaitForCacheSync waits for caches to populate.  It returns true if it was successful, false
 // if the controller should shutdown
```

Comment:
I'd add a comment to this method indicating callers should generally prefer `WaitForNamedCacheSync`